pkg/kubelet/cm/dra/state/claiminfo.go

```diff
@@ -14,111 +14,75 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package dra
+package state
 
 import (
 	"fmt"
-	"sync"
 
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/sets"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 )
 
 // claimInfo holds information required
 // to prepare and unprepare a resource claim.
-type claimInfo struct {
-	sync.RWMutex
-
-	// name of the DRA driver
-	driverName string
-
-	// claimUID is an UID of the resource claim
-	claimUID types.UID
-
-	// claimName is a name of the resource claim
-	claimName string
-
-	// namespace is a claim namespace
-	namespace string
-
-	// podUIDs is a set of pod UIDs that reference a resource
-	podUIDs sets.Set[string]
-
-	// cdiDevices is a list of CDI devices returned by the
-	// GRPC API call NodePrepareResource
-	cdiDevices []string
+type ClaimInfo struct {
+	ClaimInfoState
 
 	// annotations is a list of container annotations associated with
 	// a prepared resource
-	annotations []kubecontainer.Annotation
+	Annotations []kubecontainer.Annotation
 }
 
-func (res *claimInfo) addPodReference(podUID types.UID) {
-	res.Lock()
-	defer res.Unlock()
-
-	res.podUIDs.Insert(string(podUID))
+func (res *ClaimInfo) addPodReference(podUID types.UID) {
+	res.PodUIDs.Insert(string(podUID))
 }
 
-func (res *claimInfo) deletePodReference(podUID types.UID) {
-	res.Lock()
-	defer res.Unlock()
-
-	res.podUIDs.Delete(string(podUID))
+func (res *ClaimInfo) deletePodReference(podUID types.UID) {
+	res.PodUIDs.Delete(string(podUID))
 }
 
 // claimInfoCache is a cache of processed resource claims keyed by namespace + claim name.
 type claimInfoCache struct {
-	sync.RWMutex
-	claimInfo map[string]*claimInfo
+	claimInfo map[string]*ClaimInfo
 }
 
 // newClaimInfoCache is a function that returns an instance of the claimInfoCache.
 func newClaimInfoCache() *claimInfoCache {
 	return &claimInfoCache{
-		claimInfo: make(map[string]*claimInfo),
+		claimInfo: make(map[string]*ClaimInfo),
 	}
 }
 
-func (cache *claimInfoCache) add(claim, namespace string, res *claimInfo) error {
-	cache.Lock()
-	defer cache.Unlock()
+func generateKey(claimName, namespace string) string {
+	return claimName + namespace
+}
 
-	key := claim + namespace
+func (cache *claimInfoCache) add(res *ClaimInfo) error {
+	key := generateKey(res.ClaimName, res.Namespace)
 	if _, ok := cache.claimInfo[key]; ok {
-		return fmt.Errorf("claim %s, namespace %s already cached", claim, namespace)
+		return fmt.Errorf("claim %s, namespace %s already cached", res.ClaimName, res.Namespace)
 	}
 
-	cache.claimInfo[claim+namespace] = res
+	cache.claimInfo[key] = res
 
 	return nil
 }
 
-func (cache *claimInfoCache) get(claimName, namespace string) *claimInfo {
-	cache.RLock()
-	defer cache.RUnlock()
-
-	return cache.claimInfo[claimName+namespace]
+func (cache *claimInfoCache) get(claimName, namespace string) *ClaimInfo {
+	return cache.claimInfo[generateKey(claimName, namespace)]
```

Comment:
I don't think any of this logic belongs in this package....
This should be up in the main `dra` package, with this package being solely responsible for saving / restoring the checkpointed state.